client/state/sites/selectors.js

```diff
@@ -517,3 +518,95 @@ export function isCurrentSitePlan( state, siteId, planProductId ) {
 
 	return sitePlan.product_id === planProductId;
 }
+
+export function siteCanUpdateFiles( state, siteId ) {
+	if ( ! siteId ) {
+		return null;
+	}
+
+	if ( ! isJetpackSite( state, siteId ) ) {
+		return null;
+	}
+
+	if ( ! siteHasMinimumJetpackVersion( state, siteId ) ) {
+		return false;
+	}
+
+	const isMultiNetwork = getSiteOption( state, siteId, 'is_multi_network' );
+
+	if ( isMultiNetwork ) {
+		return false;
+	}
+
+	if ( ! isMainNetworkSite( state, siteId ) ) {
+		return false;
+	}
+
+	const fileModDisabled = getSiteOption( state, siteId, 'file_mod_disabled' );
+
+	if (
+		fileModDisabled &&
+		(
+			-1 < fileModDisabled.indexOf( 'disallow_file_mods' ) ||
+			-1 < fileModDisabled.indexOf( 'has_no_file_system_write_access' )
+		)
+	) {
+		return false;
+	}
+
+	return true;
```

Comment:
I think you could remove the condition and simplify to a single `return`:

```js
return ! includes( fileModDisabled, 'disallow_file_mods' ) && ! includes( fileModDisabled, 'has_no_file_system_write_access' );
```

Or:

```js
return ! intersection( fileModDisabled, [ 'disallow_file_mods', 'has_no_file_system_write_access' ] ).length;
```